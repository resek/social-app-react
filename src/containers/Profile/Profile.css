.ChangePassword {
    margin-top: 10px;
}