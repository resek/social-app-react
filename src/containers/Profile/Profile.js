import React, {Component, Fragment} from 'react';
import axios from "../../axios.instance";
import {Link, Redirect} from "react-router-dom";
import "./Profile.css";

class Profile extends Component {

    state = {
        logedinUserData: null,
        newPassword: null,
        redirect: false,
    }

    componentDidMount() {
        axios.get("/me")
            .then(response => {
                this.setState({logedinUserData: response.data}); 
            })
            .catch(error => {
                console.log(error);
            });        
    }

    handleInputData = (param) => (e) => {
        this.setState({[param]: e.target.value});
    }

    changePassword = (e) => {
        e.preventDefault();
        axios.patch("/me/update-password", {
            newPassword: this.state.newPassword,
            userId: this.state.logedinUserData.userId,
        })
        .then(response => {
            alert(response.data.message);
            this.setState({redirect: true});
        })
        .catch(error => {
            console.log(error);
        })
    }

    render() {

        let logedinUser;
        const data = this.state.logedinUserData;

        if (this.state.redirect) {
            return <Redirect to="/" />;
        }

        if (data) {
            logedinUser = (
                <Fragment>
                    <div>{data.username}</div>
                    <div>{data.email}</div>
                </Fragment>
            )
        }

        return (
            <Fragment>
                <div className="Navbar">
                    <Link to="/" className="Link"><button>Home</button></Link>           
                </div>
                {logedinUser}
                <form>
                    <input className="ChangePassword" placeholder="change password" type="password" onChange={this.handleInputData("newPassword")}></input>
                    <button onClick={this.changePassword}>Change password</button>
                </form>                
            </Fragment>
        )
    }    
}

export default Profile;