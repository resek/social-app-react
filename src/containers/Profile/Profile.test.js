import React from 'react';
import {shallow} from 'enzyme';

import Profile from './Profile';

test("renders component correctly", () => {
    const wrapper = shallow(<Profile />);
    expect(wrapper).toMatchSnapshot();  
});

test("component renders input", () => {
    const wrapper = shallow(<Profile />);
    expect(wrapper.find("input").length).toEqual(1);
});

test("input change test", () => {
    const wrapper = shallow(<Profile />);
    expect(wrapper.state().newPassword).toEqual(null);
    wrapper.find("input").simulate("change", {target: {value: "abcd1234"}});
    expect(wrapper.state().newPassword).toEqual("abcd1234");
})

test("componentDidMount set state with fetched data", async () => {
    const wrapper = shallow(<Profile />);
    expect(wrapper.state().logedinUserData).toBe(null);
    wrapper.instance().componentDidMount()
    await new Promise((resolve) => setTimeout(resolve, 0));
    wrapper.update();
    expect(wrapper.state().logedinUserData).toEqual({
        email: "toni@gmail.com",
        exp: 1539878266,
        iat: 1539874666,
        userId: "5bc069adb94d060015666528",
        username: "Toni"
    });
});

