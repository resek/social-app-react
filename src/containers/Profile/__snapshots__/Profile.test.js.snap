// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
<Fragment>
  <div
    className="Navbar"
  >
    <Link
      className="Link"
      replace={false}
      to="/"
    >
      <button>
        Home
      </button>
    </Link>
  </div>
  <form>
    <input
      className="ChangePassword"
      onChange={[Function]}
      placeholder="change password"
      type="password"
    />
    <button
      onClick={[Function]}
    >
      Change password
    </button>
  </form>
</Fragment>
`;
